pkg/security/grantvmgroupaccess_test.go

```diff
@@ -0,0 +1,106 @@
+package security
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+// TestGrantVmGroupAccess verifies for the three case of a file, a directory,
+// and a file in a directory that the appropriate ACEs are set, including
+// inheritance in the second two examples. These are the expected ACES. Is
+// verified by running icacls and comparing output.
+//
+// File:
+// S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704:(R,W)
+// S-1-5-83-1-3166535780-1122986932-343720105-43916321:(R,W)
+//
+// Directory:
+// S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704:(OI)(CI)(R,W)
+// S-1-5-83-1-3166535780-1122986932-343720105-43916321:(OI)(CI)(R,W)
+//
+// File in directory (inherited):
+// S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704:(I)(R,W)
+// S-1-5-83-1-3166535780-1122986932-343720105-43916321:(I)(R,W)
+
+func TestGrantVmGroupAccess(t *testing.T) {
+	f, err := ioutil.TempFile("", "gvmgafile")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
+	defer os.Remove(f.Name())
+
+	d, err := ioutil.TempDir("", "gvmgadir")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(d)
+
+	find, err := os.Create(filepath.Join(d, "find.txt"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := GrantVmGroupAccess(f.Name()); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := GrantVmGroupAccess(d); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := GrantVmGroupAccess(find.Name()); err != nil {
```

Comment:
Shouldn't `find.txt` get this for free. IE don't call this on `find.Name()` but still verify that it is set?

Reply:
Yes, I'm an idiot and got carried away with cut and paste. Fixed. Push coming soon.